src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -120,4 +120,26 @@ export function findOccurencesOf(document: vscode.TextDocument, keyword: string)
     return ranges
 }
 
+const HIGHLIGHTER = vscode.window.createTextEditorDecorationType({
+    backgroundColor: new vscode.ThemeColor('list.focusHighlightForeground'),
+})
+
+export async function highlightDocument(registry: LogStreamRegistry, uri: vscode.Uri): Promise<void> {
+    const textEditor = registry.getTextEditor(uri)
+    const logData = registry.getLogData(uri)
```

Comment:
Seems like highlightDocument() should just live on the `registry` object. IIRC, we decided that it's not really a "registry" but more like a manager or something? It's a singleton where CWL streams/searches are executed and presented.

Reply:
Yeah I was trying not to overload the registry, but I agree that it seems like the most natural place for it. Best not to force it elsewhere.

Reply:
If we merged the codelens provider w/ the document provider then `highlightDocument` would fit there.